packages/network/src/network.js

```diff
@@ -1426,3 +1426,46 @@ export function prepareLoopbackProtocolHandler(zone, { watch, allVows }) {
 
   return makeLoopbackProtocolHandler;
 }
+
+/**
+ *
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {ReturnType<import('@agoric/vow').prepareVowTools>} powers
+ */
+export const preparePortAllocator = (zone, { watch }) =>
+  zone.exoClass(
+    'PortAllocator',
+    M.interface('PortAllocator', {
+      allocateIBCPort: M.callWhen().returns(Shape.Vow$(Shape.Port)),
+      allocateICAControllerPort: M.callWhen().returns(Shape.Vow$(Shape.Port)),
+      allocateIBCPegasusPort: M.callWhen().returns(Shape.Vow$(Shape.Port)),
+      allocateLocalPort: M.callWhen().returns(Shape.Vow$(Shape.Port)),
+    }),
+    ({ protocol }) => ({ protocol, lastICAPortNum: 0n }),
+    {
+      allocateIBCPort() {
+        const { state } = this;
+        // Allocate an IBC port with a unique generated name.
+        return watch(E(state.protocol).bindPort(`/ibc-port/`));
+      },
+      allocateICAControllerPort() {
+        const { state } = this;
+        state.lastICAPortNum += 1n;
+        return watch(
+          E(state.protocol).bindPort(
+            `/ibc-port/icacontroller-${state.lastICAPortNum}`,
+          ),
+        );
+      },
+      allocateIBCPegasusPort() {
+        const { state } = this;
+        // Allocate a Pegasus IBC port with a unique generated name.
```

Comment:
Looks like there's no longer anything "unique" or "generated" about this port.
```suggestion
        // Allocate the singleton Pegasus IBC port.
```